### DISCUSSION TITLE
教程格式讨论

### DISCUSSION THREAD

**Discussion by EwTE-H** [Physics365/991CN-X-CW-Decompilation#25] [General]
上下标、粗细体等等，用一般字体难以区分，对于读者也不易理解

**Comment by Physics365** [comment]
那这样的话使用Markdown就有点不方便了，建议使用Word，最后转成Markdown和PDF，因为Word使用字体方便，对更多人也友好一些。
你觉得如何？

并且还需要统一一下教程的格式（如正文字体，缩进，排版等）

**Comment by EwTE-H** [comment]
markdown内嵌html或者纯html可以吗

**Comment by Physics365** [comment]
可以的

**Comment by Physics365** [comment]
看一下temp.pdf，这个排版可以的话就按这个来吧。
修改方法：
* 安装“模拟器及资源/字体/文档字体/static”中的所有字体按键字体和显示字体
* 更改word.docx
* 打印，打印机选择Microsoft print to PDF(其他方法格式会乱)
![屏幕截图 2024-04-03 211351](https://github.com/Physics365/991CN-X-CW-Decompilation/assets/154725076/c9018dbd-ba3a-46f5-b735-f625a75bee42)
![屏幕截图 2024-04-03 211405](https://github.com/Physics365/991CN-X-CW-Decompilation/assets/154725076/d33fa965-ee54-400a-9c01-30a497ef9a01)

**Comment by Physics365** [comment]
以上是第一章的部分，这个排版如何？ @EwTE-H

**Comment by Wenti-D** [comment]
Noto Sans 默认行距过宽，中文标准行距应当是 1.2 倍字高（不要信 Word 的段落设置，要取消「对齐到文档网格」才行），现在这是最大的问题。

**Comment by Wenti-D** [comment]
小四字号对于目前的内容来说可能太大，考虑改成 10 号（即五号）。

**Comment by Physics365** [comment]
![屏幕截图 2024-04-03 213203](https://github.com/Physics365/991CN-X-CW-Decompilation/assets/154725076/e93b3481-2022-4652-a8de-b81504c18308)
这样呢

**Comment by Wenti-D** [comment]
好些。

下一步是，章节标题、表格等特殊内容需要与上下文间有一定间距：图表上下各大约半行；节标题下大约一行，上大约一点五行；章标题适当更宽些。

**Comment by Physics365** [comment]
收到

**Comment by Wenti-D** [comment]
另外还有一些小问题：对齐方式设置为「两端对齐」而不是「左对齐」，目录应该不算第零章，以及考虑添加页码什么的，考虑多级目录什么的。

**Comment by Physics365** [comment]
这个我好像不全会弄，你看一下新增的zip包，你改一下吧（Word新手求助）😭😭😭

**Comment by Wenti-D** [comment]
呀，说实话，我是很少用 Word 排版的，因为调起来太复杂了，我要排版的话会用排版程序直接从源码生成 PDF 的……😂

**Comment by Physics365** [comment]
word里面我根本找不到某一级多级列表的行间距选项……
那这样的话应该怎么排版呢

**Comment by Wenti-D** [comment]
我本人排版用的是 LaTeX，不过要用 Word 的话当然也可以调，需要定义完善的样式，我得找个时间看看。

**Comment by Physics365** [comment]
感谢，静等更新

**Comment by EwTE-H** [comment]
像FD 20这样的十六进制编码还是用一般的代码字体吧，只有对应“屏幕显示”的部分才用卡西欧字体

**Comment by ljm673** [comment]
字符编码写成这样怎么样？
![image](https://github.com/Physics365/991CN-X-CW-Decompilation/assets/78750425/21a8ba21-34e7-4623-8378-2413743026e5)

**Comment by Physics365** [comment]
我现在觉得用正常的ABCDEF即可，大家应该看得懂的

**Comment by Wenti-D** [comment]
> 字符编码写成这样怎么样？ ![image](https://private-user-images.githubusercontent.com/78750425/319234543-21a8ba21-34e7-4623-8378-2413743026e5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIxNTY4MjIsIm5iZiI6MTcxMjE1NjUyMiwicGF0aCI6Ii83ODc1MDQyNS8zMTkyMzQ1NDMtMjFhOGJhMjEtMzRlNy00NjIzLTgzNzgtMjQxMzc0MzAyNmU1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDAzVDE1MDIwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgwY2UzZWQ5N2M5MjY1NjEyNjk4YWQ0NDY1NDU4MjQ1MzlmZTYxODM0MDJjN2MyZDM0ODU3YTkxZTJmMjI3MDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.39mKRqEdyzmah_WHWcZJ92k-903wKm89pZDf8cPBT4s)

双字节怎么办（

**Comment by ljm673** [comment]
我是说把编码写在字符上方，而不是右边
对比一下
![image](https://github.com/Physics365/991CN-X-CW-Decompilation/assets/78750425/b6edc2ac-a9ba-4fee-8106-322268e5ad42)
至于十六进制数字，用正常的ABCDEF也可以
双字节：
![image](https://github.com/Physics365/991CN-X-CW-Decompilation/assets/78750425/b5a7f615-18b6-4848-8d74-e42d052559f4)

**Comment by EwTE-H** [comment]
> 字符编码写成这样怎么样？ ![image](https://private-user-images.githubusercontent.com/78750425/319234543-21a8ba21-34e7-4623-8378-2413743026e5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIxNTkyNjksIm5iZiI6MTcxMjE1ODk2OSwicGF0aCI6Ii83ODc1MDQyNS8zMTkyMzQ1NDMtMjFhOGJhMjEtMzRlNy00NjIzLTgzNzgtMjQxMzc0MzAyNmU1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDAzVDE1NDI0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTAyZWNjN2U2YmVmMDE4NjhmMTAxY2IwZTczMTYzMWQyMGNhMDY0ZWYyZmM5MzYxOWRmYzYzOTZmMmZmODZjNjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.dejVIoRPNH9tsXapn7g3WgisHgMHvfZ0CJFoyuYQ9zo)

不太好，第一，可能导致字符边界不清（比如你这个例子里的括号到底跟根号还是跟@就一眼看不出，碰到一坨一坨的乱码字符就更加乱了），第二，编码太小了略微费眼，第三，零宽度字符怎么办
另外编码部分建议使用其他字体以示区分

**Comment by Wenti-D** [comment]
> 不太好，第一，可能导致字符边界不清（比如你这个例子里的括号到底跟根号还是跟@就一眼看不出，碰到一坨一坨的乱码字符就更加乱了），第二，编码太小了略微费眼，第三，零宽度字符怎么办 另外编码部分建议使用其他字体以示区分

关于这个问题我以前也想过，等会传一张样本上来。

**Comment by Wenti-D** [comment]
图来！

![image](https://github.com/Physics365/991CN-X-CW-Decompilation/assets/73067578/61b904db-e7b7-4bd7-9259-db22d1b921c3)

**Comment by ljm673** [comment]
挺好，LaTeX写的吧
第一行字符间距可以大一点，相邻下划线间的空隙不太清楚
我对LaTeX不太熟练，不太会写

**Comment by Wenti-D** [comment]
> 第一行字符间距可以大一点，相邻下划线间的空隙不太清楚

我故意把间距设置为 1pt 了，为了防止字符间距看起来太大🤔

**Comment by Physics365** [comment]
这样挺好的呀

**Comment by Wenti-D** [comment]
问题就在于这不是 Word 弄出来的，我在 Word 里面用文本框试了一下，调成「嵌入型」和「四周型」环绕，得到的结果会是这样：
![image](https://github.com/Physics365/991CN-X-CW-Decompilation/assets/73067578/0d5a73ff-2231-42b1-b338-defe4bc0095e)
![image](https://github.com/Physics365/991CN-X-CW-Decompilation/assets/73067578/0da6c79a-4134-44c1-8d7f-a6241ae4aeaf)

似乎问题在于 Word 的行距没法自适应？

**Comment by ljm673** [comment]
可以用域+EQ指令
效果：
![image](https://github.com/Physics365/991CN-X-CW-Decompilation/assets/78750425/d72bbeca-8803-4329-8d17-71b53eae1734)
最下面是代码，选中的是域的部分

**Comment by Wenti-D** [comment]
确实可以用，不过有时候右边会多出来一截空格，不知道为什么
![image](https://github.com/Physics365/991CN-X-CW-Decompilation/assets/73067578/09101730-5942-43df-b3ef-18e1726ea2a1)

**Comment by ljm673** [comment]
每个`\(`(转义字符)会使整个域的结尾增加一个空格
改进方法：把`\(`改为`()`，再将`)`字号改为1，颜色改为白色，藏起来

**Comment by Wenti-D** [comment]
雀食可以🤣

**Comment by Wenti-D** [comment]
扔了一个 Word 版的示例到云盘里。结果还是没有办法给表格上下加间距😡

**Comment by Physics365** [comment]
征求一下意见：这个表格是否应该用所有框线？

@Wenti-D 的式样：
![屏幕截图 2024-04-05 212532](https://github.com/Physics365/991CN-X-CW-Decompilation/assets/154725076/ddb79728-1368-44ad-bf91-1379bfaaa502)
所有框线的式样：
![屏幕截图 2024-04-05 212540](https://github.com/Physics365/991CN-X-CW-Decompilation/assets/154725076/7295d9c5-0919-470e-8e2a-ecc438329100)

**Comment by ljm673** [comment]
Wenti-D的感觉好看点😊

**Comment by Physics365** [comment]
ok,那不改了

**Comment by ljm673** [comment]
话说**云盘**在哪里

**Comment by Wenti-D** [comment]
见[讨论帖](https://github.com/Physics365/991CN-X-CW-Decompilation/discussions/19#discussioncomment-9006811)

**Comment by Physics365** [comment]
还有一个问题：标题4的缩进是特意设置的吗？这空的有点长啊
你在标题四前面空了2.2cm。。。

**Comment by Wenti-D** [comment]
显然……不是，可能我忘改了，也可能是 Word 抽风了，总之标题 2 及以下全部顶格的。

**Comment by Physics365** [comment]
![屏幕截图 2024-04-05 220251](https://github.com/Physics365/991CN-X-CW-Decompilation/assets/154725076/8a77dcd4-7b71-4d05-ad24-43a926ae0474)
这是需要的效果吗？是的话我在云盘里更新一下模板

**Comment by Wenti-D** [comment]
差不多就是这样了，6 到 9 级的应该不会使用吧。然后 2 至 4 级中间的空格可以均一以下就更好了。

**Comment by Physics365** [comment]
还有就是字符码最后我觉得可以统一多加两个00。（特殊情况除外）

**Comment by Physics365** [comment]
我觉得这个2~4级空格应该不用均一，因为实际使用几乎不可能会三级标题上下出现，单独出现时空太多反而影响美观

**Comment by Wenti-D** [comment]
我的意思是，要：
```
1.1  标题
1.1.1  标题
1.1.1.1  标题
```

而不是：
```
1.1        标题
1.1.1      标题
1.1.1.1    标题
```

也不是：
```
1.1     标题
1.1.1  标题
1.1.1.1   标题
```

**Comment by Physics365** [comment]
okok

**Comment by Physics365** [comment]
我竟不知道在哪里调

**Comment by Wenti-D** [comment]
在上方【段落】→【多级列表】→【定义新的多级列表】里调。我传了个改过的上去了。

**Comment by Physics365** [comment]
第一章完成，在云盘里：新版教程.docx，但是我没改刚刚那个格式，请你帮忙改完再传一下

**Comment by Physics365** [comment]
![屏幕截图 2024-04-06 002446](https://github.com/Physics365/991CN-X-CW-Decompilation/assets/154725076/bad0a771-a9f9-4f80-a0b4-01da96112263)

**Comment by Physics365** [comment]
为什么新的模板是这样的
```
    1.1  标题2
      1.1.1  标题3
        1.1.1.1  标题4
```

**Comment by Physics365** [comment]
![屏幕截图 2024-04-06 214125](https://github.com/Physics365/991CN-X-CW-Decompilation/assets/154725076/b8bcbada-2861-4f5c-8868-71d870de9255)

**Comment by Wenti-D** [comment]
别问，问就是 Word 又抽风了，我明明把所有的左缩进全调成过 0 了，但保存为模板再从模板新建就是会这个样子。👎😡烂

**Comment by ljm673** [comment]
要把**自动调整右缩进**去掉
![image](https://github.com/Physics365/991CN-X-CW-Decompilation/assets/78750425/3ad391b1-3eb2-493e-bddb-ee1d6129502f)

**Comment by Wenti-D** [comment]
原来是这样吗

**Comment by Physics365** [comment]
并不是，rt

**Comment by Physics365** [comment]
你左侧缩进了1.4cm(2级)

**Comment by Physics365** [comment]
`要把自动调整右缩进去掉`和这个没有关系，把1.4调为0才行

**Comment by ljm673** [comment]
雀食😅

**Comment by Wenti-D** [comment]
我设置了以后，无论是保存为空模板还是将以下内容保存为模板：
![image](https://github.com/Physics365/991CN-X-CW-Decompilation/assets/73067578/ec587c23-ad93-41b0-ae69-99d8c9b2b268)

最终都会：
![image](https://github.com/Physics365/991CN-X-CW-Decompilation/assets/73067578/d86a9252-70c1-44d0-903a-d6c6f457b0fe)

我的评价是，模板功能烂中烂👎😅

**Reply by Wenti-D** [reply]
原来不仅仅是模板功能这样，只要我设置了标题空格间距，一保存再打开，BOOM。😅

**Reply by Wenti-D** [reply]
彳亍口巴，要手动设置制表位。

**Comment by Physics365** [comment]
目录的显示级别改为2吧，不然太长了
![屏幕截图 2024-04-06 234403](https://github.com/Physics365/991CN-X-CW-Decompilation/assets/154725076/eafb0c10-9337-4515-ac04-d8b6651da0c2)
![屏幕截图 2024-04-06 234323](https://github.com/Physics365/991CN-X-CW-Decompilation/assets/154725076/06aa325a-26d5-4189-acb1-0b780a81262c)
![屏幕截图 2024-04-06 234114](https://github.com/Physics365/991CN-X-CW-Decompilation/assets/154725076/54ad7b25-4832-403b-b10d-e017f3fbff3f)

**Reply by Wenti-D** [reply]
这个以内容决定吧，甚至可以分两栏（如果一行够短的话）。

（啊哈，现在就可以这样回复了）

**Reply by ljm673** [reply]
例题前面没必要编号吧
现在是这样：
```
1.5.1  例1
         xxxxxx
1.5.2  例2
         xxxxxx
```
是不是改成
```
例1  xxx
例2  xxx
```
“例”后面的数字就个是编号啊

**Reply by Wenti-D** [reply]
就我个人经验而言，题号不会写在目录里面，用的有序列表。
![image](https://github.com/Physics365/991CN-X-CW-Decompilation/assets/73067578/b3378bc6-2ea2-446d-93d7-e9bf2fd4f686)

**Comment by Wenti-D** [comment]
修了一下，模板已传云盘 `/教程` 文件夹。

**Comment by Physics365** [comment]
收到感谢修改

**Comment by ljm673** [comment]
这里
![image](https://github.com/Physics365/991CN-X-CW-Decompilation/assets/78750425/492fd27d-4f78-4773-8d6c-e7f2c43b7994)
应该为F
其他应该没问题

**Comment by Physics365** [comment]
已修改。
到此第一章完成🎉🎉🎉

**Comment by ljm673** [comment]
排了下第三章
如图
![image](https://github.com/Physics365/991CN-X-CW-Decompilation/assets/78750425/8d665b07-b7f1-4367-8aa0-93e63dfe081f)
> 不排第二章是因为上面的图片目录里有第二章了，避免重复，留给你们吧

到时候合并上去章节编号就自动改好了
云盘修好了再传文档

**Reply by EwTE-H** [reply]
x和y字体错误

**Reply by ljm673** [reply]
已修改。文件已上传至lfs仓库，等待merge。

**Reply by ljm673** [reply]
你们第二章排了没？没的话我可以试试😉

**Reply by Physics365** [reply]
前几天我排了，但是居然没保存（大悲）
现在第二章除了上次发出的目录图什么都没有
你可以参照目录排一下，我过几天要期中考，没时间

**Reply by Wenti-D** [reply]
时时 Ctrl S，勿使己叹唉。😂

**Reply by Physics365** [reply]
qs,吸取教训

**Reply by fish4terrisa-MSDSM** [reply]
如果是word的话可以使用libreoffice，它的自动保存相当可靠的说（至少在archlinux上是这样的说）
如果是markdown或者latex的话好像很多编辑器都有自动保存的说

**Comment by Wenti-D** [comment]
嗯，看了一下目前的内容，似乎还有一些可以改进的地方：
- 有些表述看得有点云里雾里的，可以再做斟酌，例如「插入」一节：
  > 先讲插入。假设现在字符串的总长度是L，从左侧第一个字符开始计算，到从左向右第一个00为止，不包括这个00；记最左侧坐标为0，每向右一个字节坐标加一，则光标的坐标为P，字节位置算法同理，最左侧的字节是第零个字节；插入的字符长度N。（如果在框上插入，因为插完后框就没了，所以N要减一）那么会进行三个操作：
  > 1. 第L+N个字节设为00
  > 2. 把[P, L)移动到[P+N, L+N)，空出[P, P+N)
  > 3. [P, P+N)设为要插入的字符，光标移动到P+N

  改成这样可能会好些：

  > 插入操作并不像我们感受中那么直观，它涉及到三个参数：字符串总长 $L$，即从输入区开始到第一个 `00` 代码之前（不含 `00`）的总长度；光标坐标 $P$，即光标到输入区最开头之间的字节数目；插入字符长度 $N$（但若插入位置为框，则处理前两步时需减 1，因为插入时框会消失）。为描述方便，记最开头的字节为第 0 字节，那么一次完整的插入操作分为这三个步骤：
  > 1. 将第 $L + N$ 个字节置为 `00`；
  > 2. 将 $[P, L)$ 部分的字节后移 $N$ 字节，即移到 $[P+N, L+N)$ 区域；
  > 3. 将待插入的 $N$ 字节字符放到 $[P, P+N)$ 区域，光标移动到 $P+N$ 位置。

- 有些分节太细，没有必要，例如「3.2.1 历史记录在内存中的位置」里面就只有一句话，这块可能可以改成：
  > 3.1 变量
  > 　3.1.1 格式
  > 　3.1.2 合法检测
  > （甚至 3.1 可以不用再分小节，反正就一页。）
  > 3.2 历史记录
  > 3.3 两者重叠与交互

- （可考虑）修改练习/例题体系：在正文中穿插例题，在章末留下练习，在附录留下练习解答。

~本来还想说的有「更完善的例题讲解形式」，但那还是对排版要求太高了~

**Reply by ljm673** [reply]
要改的内容我觉得可以先在  #24 中提出来
> ~更完善的例题讲解形式~

"完善"具体指什么?

**Reply by Wenti-D** [reply]
> 要改的内容我觉得可以先在 #24 中提出来

确实。

> "完善"具体指什么?

计算器操作要反映的信息有三块：按键、显示屏与内存，即输入、现象、内部三个维度。在进行一项操作的时候应该可以一起将这三个维度的信息都描绘出来。例：
![image](https://github.com/Physics365/991CN-X-CW-Decompilation/assets/73067578/4eb1dfad-ee80-4359-a797-6385da3ccdb2)
一大步一个这样的展示应该就足够了。

**Comment by qiufuyu123** [comment]
要不要考虑加一点插图？
可以是实机运行的截图或者是模拟器截图，可能读者理解起来更方便
现在纯文字描述的话有点晦涩？

**Reply by Wenti-D** [reply]
有理，图文并茂才是好教程。其实不止是运行截图，有些概念的讲解也可以通过插图来辅助理解。

**Reply by fish4terrisa-MSDSM** [reply]
不过建议图片不要放太大的说，毕竟有人（我）要打印教程（方便没网的时候看），过大的话蛮费墨的说（而且容易出现打印的问题，比如出现过大墨点与未干墨迹之类的说）

**Reply by Wenti-D** [reply]
> 过大的话蛮费墨的说

费不费墨还有一个重要方面是颜色深浅，小而浓黑的图可能比大却白亮的图费的墨更多。

> （而且容易出现打印的问题，比如出现过大墨点与未干墨迹之类的说）

这要看你用的是什么打印机了，如果是喷墨，的确会有墨点不干而粘连的问题；但如果是激光，因为它用的是墨粉，所以不存在「干」这一说。

**Reply by Physics365** [reply]
一部分实机运行的截图或许可以用casio显示字体代替

**Comment by qiufuyu123** [comment]
我是这么想的
等我们完成所有章节，我们再设计一个封面什么的，然后印成几本书
印书的话价格不算很贵，淘宝几十块钱就能印很多本了，然后算是留个实体纪念？

**Reply by Physics365** [reply]
可以，我现在开始重写第一章，加入了插图，更改了例题&习题的架构
